bin/picom-trans

```diff
@@ -102,54 +106,56 @@ parse_args()
     i=$(( i + 1 ))
   done
 
-
-  while test $# -gt 0; do
-    # 15 +15 -15 are opacity values
-    # filter them out here, otherwise getopts will complain
-    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$'; then
-      target_opacity=$1
-      shift
-      continue
+  # Iterate over every argument and check if it is an opacity without the -o
+  # option in the previous argument. If so, then prepend the -o option.
+  # e.g. Turn this -
+  #     picom-trans -c +10 -s
+  # into this -
+  #     picom-trans -c -o +10 -s
+  i=1
+  while [ $i -le $# ]
+  do
+    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$' && \
+      ! eval "echo \${$#}" | grep -q '^-[hdtrgsnwc]*o$'
```

Comment:
> I'd be more comfortable with eval echo \"\${$#}\""

You mean `eval echo "\${$#}"`, right?  

> This is a greedy check and prevents selecting a window with numeric name (or id). I don't think we should touch arguments for `-n` and `-w`

Right. Good catch. Will fix.

Reply:
Should be fixed in latest commit.

Reply:
This would be even better with `printf`:

<details>
<summary>expand diff</summary>

```diff
--- a/bin/picom-trans
+++ b/bin/picom-trans
@@ -118,8 +118,8 @@ parse_args()
     # should NOT be turned into this -
     #     picom-trans -w -o 75 -o 90
     if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$' && \
-      ! eval echo "\"\${$#}\"" | grep -q '^-[hdtrgsc]*[own]$'
-    # NOTE: eval echo \${$#} means 'echo the last argument'
+      ! eval "printf '%s\n' \"\${$#}\"" | grep -q '^-[hdtrgsc]*[own]$'
+    # NOTE: eval printf '%s\n' "\${$#}" means 'echo the last argument'
     # NOTE: The letters inside the first  square brackets (ie. hdtrgsc) are
     #       the same as those in the getopts argument, minus those that are
     #       followed by a ':'
```

</details>

Currently, most implementations of `echo` interpret `echo "-n"` as expected with `-n` being the argument to `echo` without a way to indicate the end of options with `--`. We on the other hand want to echo the literal `-n` in our opacity-argument check (e.g. in case of `picom-trans -n 20`).